api/apis/auth_aware_handler.go

```diff
@@ -33,3 +35,34 @@ func (wrapper *AuthAwareHandlerFuncWrapper) Wrap(delegate AuthAwareHandlerFunc)
 		delegate(authInfo, w, r)
 	}
 }
+
+type AuthAwareHandlerFunc__NewStyle func(authInfo authorization.Info, r *http.Request) (int, interface{}, error)
+
+type AuthAwareHandlerFuncWrapper__NewStyle struct {
+	logger logr.Logger
+}
+
+func NewAuthAwareHandlerFuncWrapper__NewStyle(logger logr.Logger) *AuthAwareHandlerFuncWrapper__NewStyle {
+	return &AuthAwareHandlerFuncWrapper__NewStyle{logger: logger}
+}
+
+func (wrapper *AuthAwareHandlerFuncWrapper__NewStyle) Wrap(delegate AuthAwareHandlerFunc__NewStyle) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Add(headers.ContentType, "application/json")
```

Comment:
What should we do when the header is not JSON?
We have one or two edge cases, like the jobs endpoints for example

Reply:
One option would be to add a method to all of the JSON-encoded presenter structs, like `AsJSONEncodable()`. We could then use a type assertion in the wrapper to determine if it should be encoded as JSON, YAML (based on presence of `AsYAMLEncodable()` method), or as plain text (if it's Readable, a string or a byte slice).

Another, potentially simpler option would be to use a different wrapper method for different response types.

Reply:
Jobs endpoints [do return json](https://github.com/cloudfoundry/cf-k8s-controllers/blob/489b41eecfcbc08151254da2fe9c6513300ed1fa/api/apis/job_handler.go#L62), the `writeResponse` [function](https://github.com/cloudfoundry/cf-k8s-controllers/blob/489b41eecfcbc08151254da2fe9c6513300ed1fa/api/apis/shared.go#L349-L360) always encodes the result to json.

Reply:
My bad, I could have sworn that I worked on an endpoint that did not always return JSON, but I may be thinking of an older version of the code where we did something incorrectly.

Reply:
Some of the asynchronous delete handlers (for example, [app delete](https://github.com/cloudfoundry/cf-k8s-controllers/blob/46c9668d4c6a74f99337888d7a2057bdaf874a03/api/apis/app_handler.go#L526-L550)) don't return anything in the body. Would it matter if these responses included a `Content-Type` header?